src-cpp/rdkafkacpp.h

```diff
@@ -1380,6 +1460,25 @@ class RD_EXPORT MessageTimestamp {
  */
 class RD_EXPORT Message {
  public:
+  /** @brief Message persistance status can be used by the application to
+   *         find out if a produced message was persisted in the topic log. */
+  enum Status {
+    /**< Message was never transmitted to the broker, or failed with
+     *   an error indicating it was not written to the log.
+     *   Application retry risks ordering, but not duplication. */
+    MSG_STATUS_NOT_PERSISTED = 0,
+
+    /**< Message was transmitted to broker, but no acknowledgement was
+     *   received.
+     *   Application retry risks ordering and duplication. */
+    MSG_STATUS_POSSIBLY_PERSISTED = 1,
+
+    /**< Message was written to the log and fully acknowledged.
+     *   No reason for application to retry.
+     *   Note: this value should only be trusted with \c acks=all. */
+    MSG_STATUS_PERSISTED =  2
```

Comment:
it's questionable, but I think doing as you are an calling successful acks=1 produce requests `PERSISTED` is ok.

Reply:
I disagree, `acks=1` should be `MSG_STATUS_POSSIBLY_PERSISTED`. Broker failures and log divergence is actually quite common in practice.